provider/terraform/virtual_fields.rb

```diff
@@ -23,6 +23,16 @@ class Terraform < Provider::AbstractCore
     # behaviour. They often don't map to underlying API fields (although they
     # may map to parameters), and will require custom code to be added to
     # control them.
+    #
+    # Virtual fields are similar to url_param_only fields in that they create
+    # a schema entry which is not read from or submitted to the API. However
+    # virtual fields are meant to provider toggles for resource behavior
+    # (eg: delete_contents_on_destroy) whereas url_param_only fields _should_
+    # be used for url construction.
+    #
+    # Both are resource level fields and do not make sense, and are also not
```

Comment:
I like resource-level fields. Do you think we should duplicate this to both `url_param_only` and `virtual_fields`?

```
$FOO fields are resource-level fields, and can only be used as top-level fields.
Nested fields with similar behaviour will likely require custom code to implement.
```